src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -431,18 +429,38 @@ private BoundExpression VisitBinaryOperator(BinaryOperatorKind opKind, MethodSym
                             promotedType = _nullableType.Construct(promotedType);
                         }
 
-                        loweredLeft = PromoteEnumOperand(left, loweredLeft, promotedType, isChecked);
-                        loweredRight = PromoteEnumOperand(right, loweredRight, promotedType, isChecked);
+                        var loweredLeft = VisitAndPromoteEnumOperand(left, promotedType, isChecked);
+                        var loweredRight = VisitAndPromoteEnumOperand(right, promotedType, isChecked);
```

Comment:
loweredLeft  and loweredRight  do not yet exist at this point. I have moved lowering of operands into VisitAndPromoteEnumOperand to avoid double lowering.